### ISSUE TITLE
Problems downloading video

### ISSUE THREAD

**Issue by rcastberg** [ep1cman/unifi-protect-backup#38]
* Unifi Protect Backup version: 0.7
* Python version: 3.9.7 
* Operating System: Arm64 Docker image, tested with pyunifiprotect v3.7 and 3.9

### Description

Connects ok to UDM pro, but has issues downloading detections:
` 2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:  Adding event 629f0f8a03516603870019d9 to queue (Current queue=1)
2022-06-07 08:43:10 [INFO]:unifi_protect_backup.unifi_protect_backup:   Backing up event: 629f0f8a03516603870019d9
2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:  Remaining Queue: 0
2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Camera: G3 Instant
2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Type: motion
2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Start: 2022-06-07T10-42-46 (1654591366.637)
2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    End: 2022-06-07T10-43-02 (1654591382.822)
2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Duration: 16.185
2022-06-07 08:43:10 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Downloading video...
2022-06-07 08:43:10 [WARNING]:unifi_protect_backup.unifi_protect_backup:            Failed download attempt 1, retying in 1s
2022-06-07 08:43:10 [ERROR]:unifi_protect_backup.unifi_protect_backup:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/unifi_protect_backup/unifi_protect_backup.py", line 500, in _backup_events
    assert isinstance(video, bytes)
AssertionError
...
2022-06-07 08:43:14 [WARNING]:unifi_protect_backup.unifi_protect_backup:            Failed download attempt 5, retying in 1s
2022-06-07 08:43:14 [ERROR]:unifi_protect_backup.unifi_protect_backup:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/unifi_protect_backup/unifi_protect_backup.py", line 500, in _backup_events
    assert isinstance(video, bytes)
AssertionError
2022-06-07 08:43:15 [WARNING]:unifi_protect_backup.unifi_protect_backup:        Download failed after 5 attempts, abandoning event 629f0f8a03516603870019d9:
`

### What I Did

```
docker run -it \
  -e UFP_USERNAME='unifi-user \
  -e UFP_PASSWORD='password' \
  -e UFP_ADDRESS='10.0.10.1' \
  -e UFP_SSL_VERIFY='false' \
  -e RCLONE_DESTINATION='unifi-protect-backup:/mnt/unifi-protect-backup/' \
  -e VERBOSITY="vvvv" \
  -v /run \
  -v '/mnt/data/unifi-protect-backup/rclone.conf':'/root/.config/rclone/rclone.conf' \
  mydocker/unifi-protect-backup:latest $1

```

I rebuilt the docker image with arm64 support so I could run it on my UDM pro.

**Comment by rcastberg** [comment]
I will give it a try later today.

**Comment by rcastberg** [comment]
Same issue from my desktop:

`2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:  Adding event 629f27a802a2660387001a6d to queue (Current queue=1)
2022-06-07 10:25:57 [INFO]:unifi_protect_backup.unifi_protect_backup:   Backing up event: 629f27a802a2660387001a6d
2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:  Remaining Queue: 0
2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Camera: G3 Instant
2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Type: motion
2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Start: 2022-06-07T12-25-40 (1654597540.465)
2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    End: 2022-06-07T12-25-49 (1654597549.994)
2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Duration: 9.529
2022-06-07 10:25:57 [DEBUG]:unifi_protect_backup.unifi_protect_backup:    Downloading video...
2022-06-07 10:25:57 [WARNING]:unifi_protect_backup.unifi_protect_backup:            Failed download attempt 1, retying in 1s
2022-06-07 10:25:57 [ERROR]:unifi_protect_backup.unifi_protect_backup:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/unifi_protect_backup/unifi_protect_backup.py", line 500, in _backup_events
    assert isinstance(video, bytes)
AssertionError
2022-06-07 10:25:58 [WARNING]:unifi_protect_backup.unifi_protect_backup:            Failed download attempt 2, retying in 1s
2022-06-07 10:25:58 [ERROR]:unifi_protect_backup.unifi_protect_backup:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/unifi_protect_backup/unifi_protect_backup.py", line 500, in _backup_events
    assert isinstance(video, bytes)
AssertionError
2022-06-07 10:25:59 [WARNING]:unifi_protect_backup.unifi_protect_backup:            Failed download attempt 3, retying in 1s
2022-06-07 10:25:59 [ERROR]:unifi_protect_backup.unifi_protect_backup:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/unifi_protect_backup/unifi_protect_backup.py", line 500, in _backup_events
    assert isinstance(video, bytes)
AssertionError
2022-06-07 10:26:00 [EXTRA_DEBUG]:unifi_protect_backup.unifi_protect_backup:    Checking the status of the websocket...
2022-06-07 10:26:00 [EXTRA_DEBUG]:unifi_protect_backup.unifi_protect_backup:    Websocket is connected.
2022-06-07 10:26:00 [WARNING]:unifi_protect_backup.unifi_protect_backup:            Failed download attempt 4, retying in 1s
2022-06-07 10:26:00 [ERROR]:unifi_protect_backup.unifi_protect_backup:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/unifi_protect_backup/unifi_protect_backup.py", line 500, in _backup_events
    assert isinstance(video, bytes)
AssertionError
2022-06-07 10:26:01 [WARNING]:unifi_protect_backup.unifi_protect_backup:            Failed download attempt 5, retying in 1s
2022-06-07 10:26:01 [ERROR]:unifi_protect_backup.unifi_protect_backup:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/unifi_protect_backup/unifi_protect_backup.py", line 500, in _backup_events
    assert isinstance(video, bytes)
AssertionError
2022-06-07 10:26:02 [WARNING]:unifi_protect_backup.unifi_protect_backup:        Download failed after 5 attempts, abandoning event 629f27a802a2660387001a6d:
2022-06-07 10:27:00 [EXTRA_DEBUG]:unifi_protect_backup.unifi_protect_backup:    Checking the status of the websocket...
2022-06-07 10:27:00 [EXTRA_DEBUG]:unifi_protect_backup.unifi_protect_backup:    Websocket is connected.
`
Using: 
`docker run -it \
  -e UFP_USERNAME='user' \
  -e UFP_PASSWORD='pass' \
  -e UFP_ADDRESS='10.0.10.1' \
  -e UFP_SSL_VERIFY='false' \
  -e RCLONE_DESTINATION='unifi-protect-backup:/mnt/unifi-protect-backup/' \
  -e VERBOSITY="vvvv" \
  --network vpn \
  -v /run \
  -v '/home/bla/rclone.conf':'/root/.config/rclone/rclone.conf' \
  ghcr.io/ep1cman/unifi-protect-backup `

**Comment by ep1cman** [comment]
This tool was originally developed against `v3.2.1` of pyunifiprotect, but you mention " v3.7 and 3.9". Is there a reason for this change? and could you try using 3.2.1? 

I will need to verify the latest releases of pyunifiprotect are compatible because it seems some of the code relating to downloading videos has changed

**Comment by rcastberg** [comment]
It was just because that was what was pulled in, I will try again when I
have access to the device later this week.

But for the test when I had it running on the desktop I used the image that
you provided through github.

**Comment by ep1cman** [comment]
Hi,

I have bumped to the latest version of pyunifiprotect on the `dev` branch/container tag. Please try this as it defintiely fixes some issues with 2.0.1.

It seems they have also changed the way user accounts are created and they need to be "assigned" cameras. Please check the readme on `dev` for instructions to ensure you have all the cameras assigned to the account if you are using a specific user for this tool.

**Comment by rcastberg** [comment]
Hi, this seems to have solved my issue. 
Thanks.
